opal/resource.go

```diff
@@ -281,7 +283,16 @@ func resourceResourceUpdateVisibility(ctx context.Context, d *schema.ResourceDat
 }
 
 func resourceResourceUpdateReviewers(ctx context.Context, d *schema.ResourceData, client *opal.APIClient, reviewersI any) diag.Diagnostics {
-	rawReviewers := reviewersI.([]any)
+	// reviewersI could be a schema.Set from terraform or our own constructed slice.
```

Comment:
Not sure I understand when which of these would happen. Mind elaborating a bit pls?

Reply:
there are a couple callers that use `resourceResourceUpdateReviewers` that pass a slice instead of a `*schema.Set`. (because we pass a default value instead of relying on the terraform-engine-provided value). i.e. here:
https://github.com/opalsecurity/terraform-provider-opal/blob/57f31542d305fc146280e085ce42dcf2aab2e3d1/opal/resource.go#L252

I kind of hate that this is polymorphic, but I could not figure out how to get `schema.NewSet` working in a reasonable timebox, so I opted for this hack instead.